src/sources/host_metrics/mod.rs

```diff
@@ -753,3 +753,24 @@ pub(self) mod tests {
         }
     }
 }
+
+#[cfg(feature = "host-metrics-integration-tests")]
```

Comment:
Is it really necessary to run this particular test inside a container and not just run it as a standard test ?

Reply:
Running it here locally in linux it passes fine as a unit test. My main concern was with other platforms.
I suppose I could try making it a unit test and see if CI fails though.

Reply:
This source should always work on all hosts, it'll just report different sets of metrics. So I think this should not be separated into its own integration test setup. The tests should instead be careful to rely only on those metrics that are common to all platforms (including in or out of containers).

Reply:
Moved it to unit test in https://github.com/vectordotdev/vector/pull/14103/commits/34d33cbd9b3daacdd011cba362dc2c5c5b12823e

Reply:
I don't think this needs a feature gate anymore now that it is a unit test. It can just be moved up to `mod tests`.

Reply:
Yeah I pushed too soon :sweat_smile:  ... collapsed up into the existing module.